pkg/catalog/install_test.go

```diff
@@ -163,6 +163,101 @@ status: {}
 			})
 		})
+
+		When("a url is provided with branch", func() {
+			It("generates a spec with url and branch and doesn't call show", func() {
+				cfg = catalog.InstallConfig{
+					Branch:        "main",
+					CatalogName:   "nginx",
+					CatalogClient: fakeCatalogClient,
+					Namespace:     "default",
+					ProfileName:   "nginx-1",
+					SubName:       "mysub",
+					URL:           "https://github.com/weaveworks/profiles-examples",
+					Directory:     tempDir,
+					Path:          "branch-nginx",
+				}
+				err := catalog.Install(cfg)
+				Expect(err).NotTo(HaveOccurred())
+
+				var files []string
+				profileDir := filepath.Join(tempDir, "nginx-1")
+				err = filepath.Walk(profileDir, func(path string, info os.FileInfo, err error) error {
+					files = append(files, path)
+					return nil
+				})
+				Expect(err).NotTo(HaveOccurred())
+
+				profileFile := filepath.Join(profileDir, "profile.yaml")
+				artifactFile := filepath.Join(profileDir, "kustomize-0.yaml")
+				Expect(files).To(ConsistOf(profileDir, profileFile, artifactFile))
+
+				content, err := ioutil.ReadFile(profileFile)
+				Expect(err).NotTo(HaveOccurred())
+				Expect(string(content)).To(Equal(`apiVersion: weave.works/v1alpha1
+kind: ProfileSubscription
+metadata:
+  creationTimestamp: null
+  name: mysub
+  namespace: default
+spec:
+  branch: main
+  path: branch-nginx
+  profileURL: https://github.com/weaveworks/profiles-examples
+status: {}
+`))
+
+				content, err = ioutil.ReadFile(artifactFile)
+				Expect(err).NotTo(HaveOccurred())
+				Expect(string(content)).To(Equal(`apiVersion: api
+kind: kustomize
+metadata:
+  creationTimestamp: null
+  name: foo
+  namespace: default
+spec:
+  interval: 0s
+  prune: true
+  sourceRef:
+    kind: ""
+    name: ""
+status: {}
+`))
+			})
+		})
+
+		When("a url is provided without path", func() {
+			It("returns a sensible error that path is required with a url", func() {
+				cfg = catalog.InstallConfig{
+					CatalogName:   "nginx",
+					CatalogClient: fakeCatalogClient,
+					Namespace:     "default",
+					ProfileName:   "nginx-1",
+					SubName:       "mysub",
+					URL:           "https://github.com/weaveworks/profiles-examples",
+					Directory:     tempDir,
+				}
+				err := catalog.Install(cfg)
+				Expect(err).To(MatchError("path must be provided with url"))
+			})
+		})
+		When("a branch is provided which isn't domain compatible", func() {
+			It("will return an error", func() {
+				cfg = catalog.InstallConfig{
+					CatalogName:   "nginx",
+					CatalogClient: fakeCatalogClient,
+					Namespace:     "default",
+					ProfileName:   "nginx-1",
+					SubName:       "mysub",
+					URL:           "https://github.com/weaveworks/profiles-examples",
+					Directory:     tempDir,
+					Branch:        "not_domain_compatible",
+					Path:          "path",
+				}
+				err := catalog.Install(cfg)
+				Expect(err).To(MatchError("branch must match RFC 1123 subdomain format"))
```

Comment:
whats causing this again? It seems odd to block branches that contain `_` as I imagine thats quite common

Reply:
It's added to the resource name which must be dns compliment

Reply:
mhmm, can we strip away the `_`'s and others instead of blocking it? If someones working on a feature PR they shouldn't be prevent from creating the profile because of `_`'s in the branch name

Reply:
Sure, we can convert it.

Reply:
The problem is that it's not easy. Like, people have all kinds of branches. And stripping out every character is bound to be failure prone. :/ What I'm saying is that it won't be a 100% fool proof coverage.

Reply:
also, things like `-branch` and `branch--branch` are invalid as well...

Reply:
> The problem is that it's not easy. Like, people have all kinds of branches. And stripping out every character is bound to be failure prone. :/ What I'm saying is that it won't be a 100% fool proof coverage.

yeah... but it feels better than blocking branches with underscores. I'm sure some tool must existing in the k8s community for sanitising names?

alternatively, can we use a different naming convention for the git repository resource that doesn't include the branch name, but is still unique?